### PR TITLE
FFT factory funcs fix and bugfix for interpolated time point thresh

### PR DESCRIPTION
**Pull request by ggmarshall**
Added check in Proc_Manager for if shape mismatch is due to buffer len being included in shape, this was causing extra dimensions to be added for the ffts factory functions causing shapes mismatch. Probably a smarter way to do this if I understood processing chain better but this works in testing.
Also added bugfix for interpolated time point thresh last pr added returns but this causes interpolating behaviour to be skipped, changed to breaks so will keep intended behaviour of having first crossing returned

### CODE REVIEW

File: src/dspeed/processing_chain.py [hunk 1]
```diff
@@ -1407,12 +1407,21 @@ def __init__(
 
                 # reshape just in case there are some missing dimensions
                 arshape = list(param.shape)
-                for idim in range(-1, -1 - len(shape), -1):
-                    if (
-                        len(arshape) < len(shape) + 1 + idim
-                        or arshape[idim] != shape[idim]
-                    ):
-                        arshape.insert(len(arshape) + idim + 1, 1)
+                # check of shape mismatch is just due to wf buffer dim being included
```
Comment (iguinn) [inline src/dspeed/processing_chain.py:1410]: I suggest reverting this change, and instead adding after line 1263:
`if not param.is_const: arr_dims += [self.proc_chain._bock_width]`

**Comment by iguinn** [comment]
Thanks for catching and investigating this! Also thanks for the bug fix to the interpolated time point threshold

As discussed on slack, this is probably caused by changes I made to allow const parameters, where I wanted to stop always assuming the outer dimension was equal to the waveform block length. This works except when the outer block width is expected to be a part of the function signature; this is the case in the FFTW processors because they are able to optimize better if they get the full waveform block rather than one at a time. The above suggested fix should work; the proposed fix I think leaves more room for edge cases.

In addition, can you add a json processor block for one of the FFTW processors to the integration test config file, so that something like this gets caught in the future? The existing fftw tests are independent of the processing chain machinery.

File: src/dspeed/processing_chain.py (hunk 1 above)
Comment (ggmarshall) [reply src/dspeed/processing_chain.py:1410]: This doesn't seem to work in testing as the arshape doesn't have the block_width either

Comment (ggmarshall) [reply src/dspeed/processing_chain.py:1410]: so it breaks it for all the other processors

Comment (iguinn) [reply src/dspeed/processing_chain.py:1410]: Oh yeah, the block width dim has to go first, try `arr_dims = [self.proc_chain._bock_width] + arr_dims` instead
